python/deptracker.py

```diff
@@ -0,0 +1,81 @@
+from itertools import imap, chain
+from functools import wraps
+
+__all__ = ("deptracker",)
+
+class deptracker(object):
+    """
+    Basic dependency tracking for member methods of an object, using a decorator class
+
+    The member methods can e.g. be decorated as follows:
+    >>> @deptracker(before=("create", "eleSmear"), after="eleReg", performs=("eleSmear", "correction"))
+    >>> def applyElectronSmearing(self):
+    >>>     ...
+
+    which will, when calling this method, check that no member method with `@deptracker(..., performs=("create"))`
+    nor `@deptracker(..., performs=("eleSmear"))` has been called yet, and
+    that a method with `@deptracker(..., performs=("eleReg", ...))` has been called.
+    Furthermore, this will mark "eleSmear" and "correction" as "done", such that no other method
+    with either of these in "before" can be called after (for "eleSmear": including this method;
+    putting the same tag in "before" and "after" ensures that a method is called no more than once).
```

Comment:
same here in documentation, I would prefer explicit naming of the "tags" rather than abbreviations

Reply:
ok